lib/Dialect/TritonGPU/Transforms/Pipeliner/Schedule.cpp

```diff
@@ -190,15 +191,50 @@ tt::CoarseSchedule::createFinalSchedule(scf::ForOp forOp) const {
 }
 
 void tt::CoarseSchedule::dump() {
-  assert(numStages > 0 && "Invalid number of stages");
-  for (int i = 0; i < numStages; i++) {
-    llvm::dbgs() << "\n---- Ops in stage " << i << "\n";
-    for (auto &[op, stageAndCluster] : opToStageAndCluster) {
-      if (i == stageAndCluster.first) {
-        llvm::dbgs() << "        cluster: " << *stageAndCluster.second
-                     << ":\n\t" << *op << "\n";
-      }
+  auto getOpSortIdent = [](Operation *op) {
```

Comment:
can we just make `opToStageAndCluster` a MapVector?

Reply:
d'oh

that works but you have drop `const` in a few members of `CoarseSchedule` - you cool with that? Also I'll note the intra stage ordering (clusters) will still be odd, not necessarily sorted (but the dump will ofc be deterministic).